### PR TITLE
Adding setuptools==40.4.3

### PR DESCRIPTION
**Pull request by ghost**
setuptools==40.4.3

### CODE REVIEW

**Comment by cculianu** [comment]
What happens if it's missing? Things seemed to work before.. :)

(Note: I am lacking in the detailed knowledge with some of this python setup stuff... so I just want to be educated on this. )....

**Comment by SomberNight** [comment]
This is a generated file created by [freeze_packages.sh](https://github.com/Electron-Cash/Electron-Cash/blob/20e915b136399c06cc2c8b503deb8d5758fd89d7/contrib/freeze_packages.sh) based on [contrib/requirements/requirements.txt](https://github.com/Electron-Cash/Electron-Cash/blob/20e915b136399c06cc2c8b503deb8d5758fd89d7/contrib/requirements/requirements.txt)

**Comment by cculianu** [comment]
@SomberNight does that mean there's no point in updating it?  So many questions I have.. :)

**Comment by SomberNight** [comment]
not sure but I don't think so. OP should say why he added it.
but when adding a new top level dependency, it should not be added into this file, rather into `contrib/requirements/requirements.txt` (or -hw or -binaries), and then this file should be regenerated.

(although I would note that the ElectronCash windows binaries are probably not deterministic as Electrum had to jump through some further hoops to make them so, in which case this abstraction layer is unnecessary anyway)

**Comment by cculianu** [comment]
Ah, cool. Thanks for the info. I didn't know all that but now I know what to look at next time I build (out of curiosity, again). 🗡

**Comment by wisbwond** [comment]
In `https://github.com/Electron-Cash/Electron-Cash/blob/20e915b136399c06cc2c8b503deb8d5758fd89d7/contrib/build-wine/build-electrum-git.sh#L105 `  the line `python3 setup.py sdist --format=zip,gztar` is not necessary while building i think. Due to this line the below error.

```
Traceback (most recent call last):
  File "setup.py", line 5, in <module>
    from setuptools import setup
ModuleNotFoundError: No module named 'setuptools'

```

Either its better to remove the above line or install setuptools before running setup.py. correct me if i'm wrong.

**Comment by SomberNight** [comment]
> line python3 setup.py sdist --format=zip,gztar is not necessary while building i think

Yes, that builds linux source dists; not sure why it's there.

**Comment by cculianu** [comment]
So do I kill this PR or what?  **So confused...**

File: contrib/deterministic-build/requirements.txt [hunk 1]
```diff
@@ -11,3 +11,4 @@ qrcode==6.0
 requests==2.18.4
 six==1.11.0
 urllib3==1.22
+setuptools==40.4.3
```
Comment (cculianu) [inline contrib/deterministic-build/requirements.txt]: But WHy?!

**Comment by cculianu** [comment]
I'm closing this PR as the author seems to be incommunicado.

If you want this merged simply write here why this change needs to happen...

**Pull request closed without merging**